testgrid/conformance/upload_e2e.py

```diff
@@ -153,27 +158,42 @@ def testgrid_finished_json_contents(finish_time, passed, metadata):
         'result': result
     })
 
+
 def upload_string(gcs_path, text, dry):
     """Uploads text to gcs_path if dry is False, otherwise just prints"""
     cmd = ['gsutil', '-q', '-h', 'Content-Type:text/plain', 'cp', '-', gcs_path]
-    print >>sys.stderr, 'Run:', cmd, 'stdin=%s'%text
+    print >>sys.stderr, 'Run:', cmd, 'stdin=%s' % text
     if dry:
         return
     proc = subprocess.Popen(cmd, stdin=subprocess.PIPE)
     proc.communicate(input=text)
     if proc.returncode != 0:
-        raise RuntimeError("Failed to upload with exit code: %d" % proc.returncode)
+        raise RuntimeError(
+            "Failed to upload with exit code: %d" % proc.returncode)
+
 
 def upload_file(gcs_path, file_path, dry):
     """Uploads file at file_path to gcs_path if dry is False, otherwise just prints"""
-    cmd = ['gsutil', '-q', '-h', 'Content-Type:text/plain', 'cp', file_path, gcs_path]
+    cmd = ['gsutil', '-q', '-h', 'Content-Type:text/plain',
+           'cp', file_path, gcs_path]
     print >>sys.stderr, 'Run:', cmd
     if dry:
         return
     proc = subprocess.Popen(cmd)
     proc.communicate()
     if proc.returncode != 0:
-        raise RuntimeError('Failed to upload with exit code: %d' % proc.returncode)
+        raise RuntimeError(
+            'Failed to upload with exit code: %d' % proc.returncode)
+
+
+def activate_service_account(key_file, dry_run):
+    """activates a gcp service account by shelling out to gcloud"""
+    cmd = ["gcloud", "auth", "activate-service-account", "--key-file="+key_file]
```

Comment:
Do we need to de-activate when we are done?

Reply:
It depends on your infra, I think not? It's just setting the gcloud account. If you're running in a container or temporary home it should be deleted on job exit anyhow?

I can add it if anyone needs that